### PR TITLE
[stable/traefik] Add config option for hostNetwork

### PR DESCRIPTION
**Pull request by mrbalihai**
Signed-off-by: Robert Bollons <rob@robbollons.com>

#### What this PR does / why we need it:
This will allow the hostNetwork value to be configured for the deployment. It defaults to false but when set to true it will allow external ingress on non-load balanced k8s instances such as a VPSs.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/traefik/values.yaml [hunk 1]
```diff
@@ -350,6 +350,7 @@ deployment:
   #   key: value
   # podAnnotations:
   #   key: value
+  hostNetwork: false
```
Comment (dtomcej) [inline stable/traefik/values.yaml]: This should be commented out by default

File: stable/traefik/templates/deployment.yaml [hunk 2]
```diff
@@ -83,6 +83,7 @@ spec:
           periodSeconds: 10
           successThreshold: 1
           timeoutSeconds: 2
+        hostNetwork: {{ .Values.deployment.hostNetwork }}
```
Comment (dtomcej) [inline stable/traefik/templates/deployment.yaml:87]: This entire setting should be optional. You need to test if `.Values.deployment.hostNetwork` exists, then use the setting if it does. Keeps the yaml cleaner.

**Review by dtomcej — CHANGES_REQUESTED** [review]

File: stable/traefik/values.yaml (hunk 1 above)
Comment (mrbalihai) [reply stable/traefik/values.yaml]: Fixed.

**Comment by Hessu1337** [comment]
Would be great to have this parameter released

**Comment by alexpovel** [comment]
> Would be great to have this parameter released

Second this.

**Pull request closed without merging**